src/main/java/org/jenkinsci/plugins/GithubAuthenticationToken.java

```diff
@@ -504,6 +504,9 @@ private GHMyself loadMyself(@NonNull String token) throws IOException {
                 // Also stick into usersByIdCache (to have latest copy)
                 String username = ghMyself.getLogin();
                 usersByIdCache.put(username, new GithubUser(ghMyself));
+            } else {
+                // force creation of the gh variable, esp. in case of impersonation
+                getGitHub();
```

Comment:
I *think* this is the cause of the performance degradation

Reply:
@lprimak can you describe more about your environment and usage?

* In what ways to you use auth?  Just oauth web or GitHub personal access tokens, too?
* How are your jobs organized?  Are they in folders or grouped into views?  What columns do you have enabled on the landing page if Jenkins for the All view?
* Where were performance issues most noticeable?
* How many users?
* How many jobs and can you break down numbers by job type?

Hopefully, with enough info, it can be reproduced and fixed upon reproduction.  Any answers you provide can help us figure out a performance profile and attempt to emulate your environment.

Reply:
Also if you could describe how you evaluated the 20x performance reduction it will give us hints of where to look as a root cause.  It's not always just the contributed area that could be the root cause, it's possible an unrelated area of code could trigger the conditions for this performance issue.  So it would give us a hint for evaluating other areas if we understand exactly how you experienced the degradation.

Reply:
> In what ways to you use auth? Just oauth web or GitHub personal access tokens, too?

I use auth to sign in to jenkins with GH credentials, I believe that's all:
Here is my oath configuration (config-as-code)

```
jenkins:
  securityRealm:
    github:
      clientID: "617b74bd0af85c99f9bb"
      clientSecret: "{AQAAABAAAAAwCuMh5ZYz2AQdCQQKFtvtFaur2Z9vD6zf5xwV1KhHbz+cSEfDwCs2gf/P6JwzCBC4/LwkZLVYD56cXr6ZkZwevQ==}"
      githubApiUri: "https://api.github.com"
      githubWebUri: "https://github.com"
      oauthScopes: "read:org,user:email,repo"
```

> How are your jobs organized? Are they in folders or grouped into views? 

There are less than 10 jobs. No folders

> Where were performance issues most noticeable?

Running any declarative pipeline. It's slow as molasses

> How many users?

Just me

> Also if you could describe how you evaluated the 20x

Pipelines that used to take 1 minute now take 15 minutes :)

Thanks for your help